frontend/src/pages/ServicesPage/ServicesPage.tsx

```diff
@@ -45,30 +46,45 @@ export const ServicesPage = connect(mapState)(({ connections, devices }: Service
           <Breadcrumbs />
           <Typography variant="h1">
             <ConnectionStateIcon service={device} connection={activeConnection} size="lg" />
-            <ServiceName
-              service={device}
-              connection={activeConnection}
-              shared={device.shared === 'shared-from'}
-              inline
-            />
+            <ServiceName service={device} connection={activeConnection} shared={device.shared} inline />
             <DeleteButton device={device} />
           </Typography>
         </>
       }
     >
-      {/* <Typography variant="subtitle1">Services</Typography> */}
+      {searched && <Subtitle primary="Services" secondary={`Searched for “${query}”`} />}
       <ServiceList services={device.services} connections={serviceConnections} />
       <Typography variant="subtitle1">Device details</Typography>
       <Columns count={1} inset>
         <DataDisplay
           data={[
+            { label: 'Availability', value: percent(device.availability) },
+            { label: 'Instability', value: percent(device.instability) },
+            { label: 'Location', value: location(device.geo) },
+            { label: 'ISP', value: device.geo?.isp },
```

Comment:
Add connection type: One of the following values: Dialup, Cable/DSL, Corporate, or Cellular. Additional values may be added in the future.